### PR TITLE
Add move C-tor and move assignment.

### PR DESCRIPTION
**Pull request by runer53248**
Add move C-tor and move assignment. 
Mark copy C-tor and copy assignment as deleted. 
Add tests for move semantics. 
Add warning flags in cmake.

### CODE REVIEW

**Review by raxonpc — APPROVED** [review]

File: homework/unique_ptr/UniquePointer.hpp [hunk 1]
```diff
@@ -18,6 +19,19 @@ class UniquePointer {
         pointer_ = nullptr;
     }
 
+    UniquePointer(const UniquePointer&) noexcept = delete;
+    UniquePointer& operator=(const UniquePointer&) noexcept = delete;
```
Comment (michalsoloducha) [inline homework/unique_ptr/UniquePointer.hpp]: noexcept is redundant when delete is in use

**Review by michalsoloducha — COMMENTED** [review]
one comment inline

**Review by michalsoloducha — APPROVED** [review]

File: homework/unique_ptr/CMakeLists.txt [hunk 2]
```diff
@@ -15,7 +15,7 @@ add_executable(${PROJECT_NAME} main.cpp)
 add_executable(${PROJECT_NAME}-ut Tests/testsMain.cpp Tests/tests.cpp)
 target_link_libraries(${PROJECT_NAME}-ut PRIVATE gtest_main gmock_main pthread)
 
-# target_compile_options(${PROJECT_NAME}-ut PRIVATE -g)
+target_compile_options(${PROJECT_NAME}-ut PRIVATE -Wall -Wextra -pedantic)
```
Comment (raxonpc) [inline homework/unique_ptr/CMakeLists.txt:18]: We can also check, if msvc is used (to set the according /W4 flag)

File: homework/unique_ptr/CMakeLists.txt [hunk 3]
```diff
@@ -15,7 +15,7 @@ add_executable(${PROJECT_NAME} main.cpp)
...
 
 enable_testing()
 add_test(NAME UniquePtrTest COMMAND ${PROJECT_NAME}-ut --gtest_color=yes)
```
Comment (raxonpc) [inline homework/unique_ptr/CMakeLists.txt:21]: I think the name should also be dependent on the project name

**Pull request merged**